lib/mongoid/criteria/queryable/selector.rb

```diff
@@ -150,6 +150,8 @@ def evolve_multi(specs)
         #
         # @return [ Object ] The serialized object.
         def evolve(serializer, value)
+          return value.raw_value if value.is_a?(Mongoid::RawValue)
```

Comment:
This is the key line.

Note that under the current structure, we cannot define `Mongoid::RawValue#evolve`. The reason is that this method `Mongoid::Criteria::Queryable::Selector#evolve(serializer, value)` calls the class-specific `#evolve` methods based on the field-specific `serializer`, which defeats the goal of `RawValue` which is to *bypass the field-specific serialization*.

In the future, I think this line should be `raw_value.mongoize` as I have noted in MONGOID-5468.

Reply:
This looks fine, I don't see any issues here right now.